adafruit_matrixportal/matrixportal.py

```diff
@@ -243,36 +243,36 @@ def set_text(self, val, index=0):
         # Make sure at least a single label exists
         if not self._text:
             self.add_text()
-        if self._text_font:
-            string = str(val)
-            if self._text_maxlen[index]:
-                string = string[: self._text_maxlen[index]]
-            if self._text[index]:
-                # print("Replacing text area with :", string)
-                # self._text[index].text = string
-                # return
-                try:
-                    text_index = self.splash.index(self._text[index])
-                except AttributeError:
-                    for i in range(len(self.splash)):
-                        if self.splash[i] == self._text[index]:
-                            text_index = i
-                            break
-
-                self._text[index] = Label(self._text_font, text=string)
-                self._text[index].color = self._text_color[index]
-                self._text[index].x = self._text_position[index][0]
-                self._text[index].y = self._text_position[index][1]
-                self.splash[text_index] = self._text[index]
-                return
-
-            if self._text_position[index]:  # if we want it placed somewhere...
-                print("Making text area with string:", string)
-                self._text[index] = Label(self._text_font, text=string)
-                self._text[index].color = self._text_color[index]
-                self._text[index].x = self._text_position[index][0]
-                self._text[index].y = self._text_position[index][1]
-                self.splash.append(self._text[index])
+        string = str(val)
+        if not string:
+            max_glyphs = 50
+        else:
+            max_glyphs = len(string)
+        if self._text_maxlen[index]:
+            string = string[: self._text_maxlen[index]]
+        print("text index", self._text[index])
+        if self._text[index] is not None:
+            print("Replacing text area with :", string)
+            self._text[index].text = string
+            text_index = self.splash.index(self._text[index])
+
+            self._text[index] = Label(
+                self._text_font, text=string, max_glyphs=max_glyphs
+            )
+            self._text[index].color = self._text_color[index]
+            self._text[index].x = self._text_position[index][0]
+            self._text[index].y = self._text_position[index][1]
+            self.splash[text_index] = self._text[index]
+
+        elif self._text_position[index]:  # if we want it placed somewhere...
+            print("Making text area with string:", string)
+            self._text[index] = Label(
+                self._text_font, text=string, max_glyphs=max_glyphs
+            )
+            self._text[index].color = self._text_color[index]
+            self._text[index].x = self._text_position[index][0]
+            self._text[index].y = self._text_position[index][1]
+            self.splash.append(self._text[index])
```

Comment:
I think this is clearer:

```suggestion
        index_in_splash = None
        if self._text[index] is not None:
            print("Replacing text area with :", string)
            self._text[index].text = string
            index_in_splash = self.splash.index(self._text[index])

        self._text[index] = Label(
            self._text_font, text=string, max_glyphs=max_glyphs
        )
        self._text[index].color = self._text_color[index]
        self._text[index].x = self._text_position[index][0]
        self._text[index].y = self._text_position[index][1]
            
        if index_in_splash is not None:
            self.splash[text_index] = self._text[index]
        else:
            self.splash.append(self._text[index])
```

Is it correct to not update the text if no position is given? That seems like a bug.

Reply:
Ok, you know... I adapted that from the original PyPortal and it likely might be a bug that I hadn't come across yet and really only a bug because I did things a bit differently like allowing use of the terminal font. I like your suggestion and I'll go head and test it out and either commit it or tweak and commit it. Thanks.

Reply:
Ok, I added a small change but it's working, so I'm happy with it.